### DISCUSSION TITLE
two way communication with cables

### DISCUSSION THREAD

**Discussion by inst5** [cables-gl/cables_docs#170] [Answers & Questions] [answered]
Hi,
I would lile to embed a cables patch in an iFrame on my website and let the patch with the rest of the sie exchange data.

For example I have simple PLAY button that I would like to connect to an action in my calbes patch.
This line I use on my site: <a class="use-ajax" id='btPlayBlack1' href="#!"><i class="fas fa-play-circle"></i> <font size="4">Track1</font></a>

Or I would like so send the xy coordinates from my patch to an other script on my site.

I found this already on the cables-page: https://cables.gl/docs/4_export_embed/dev_embed_functions/dev_embed_functions
But is there a tutorial that could show as video HOWTO to make cables communicate?

thanks!

**Comment by positlabs** [comment, marked as answer]
I'm not sure if it will work with the hosted iframe (because of cross origin restrictions), but you could try postmessage https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage

Otherwise, you could host your own files and communicate simply by dispatching custom events on the window, then listening for them in a custom op that routes the messages to the rest of the graph.

**Comment by pandrr** [comment]
@inst5 you found the docs, it should be all in there

there is also a repo with some examples: https://github.com/cables-gl/cables-embedding

example_vars_getset shows exactly what you are asking

**Comment by inst5** [comment]
Hi,
I treid the functio to trigger with putting this inmy html:
<input type="button" value="button" onclick="CABLES.patch.config.patchFunctiontrigger();">

It works in the same html file but not with an iframe with cables patch an calling the function from outside.

**Comment by steam0r** [comment]
hey, yeah, iframes cannot communicate with each others that way for security reasons. you will have to fo with what @positlabs posted and somehow implement your own "protocol", or, if possible, not use an iframe (which can also lead to different other issues with your patch not being focussed and scrolling behaviour).

**Reply by inst5** [reply]
ok, so I will try to avoid the ifram solution.
